Calibration/IsolatedParticles/src/CaloPropagateTrack.cc

```diff
@@ -601,6 +602,27 @@ namespace spr{
     return spr::propagateECAL (vertex, momentum, charge, bfield, debug);
   }
 
+  std::pair<DetId,bool> propagateIdECAL(const HcalDetId& id, const CaloGeometry* geo, const MagneticField* bField, bool debug) {
+
+    const HcalGeometry* gHB = (const HcalGeometry*)(geo->getSubdetectorGeometry(DetId::Hcal,HcalBarrel));
+    GlobalPoint vertex(0,0,0);
```

Comment:
hi @bsunanda - could this be declared const?

Reply:
Added const to all GlobalPoint, GlobalVector where applicable